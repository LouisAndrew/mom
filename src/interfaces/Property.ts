import { FluidObject } from 'gatsby-image';

export type Property = {
    name: string;
    address: string;
    propertyType: 'house' | 'apartment' | 'kavling' | 'home-office';
    saleType: 'sell' | 'rent';

    // optional fields..
    location?: string;
    imgs?: { image: string | FluidObject | FluidObject[]; imgAlt: string }[];
    description?: string;
    isTrending?: boolean;
    geolocation?: { lat: number; lng: number };
    price?: number;
    area?: number;
    roomQty?: number;
    bathQty?: number;
};
