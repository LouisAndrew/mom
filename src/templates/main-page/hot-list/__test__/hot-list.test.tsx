import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import HotList from '..';
import { Property } from 'interfaces/Property';

describe('Hot List component', () => {
    const mockProperties: Property[] = [
        {
            name: 'Rumah 1',
            address: 'jalan Mangga 24 blok G nomor 167',
            imgs: [
                {
                    image:
                        'https://res.cloudinary.com/dsvdffre0/image/upload/v1595340453/ovmhcixqqpfjla4mxovf.jpg',
                    imgAlt: 'Handsome boy',
                },
            ],
            porpertyType: 'house',
            saleType: 'sell',
        },
        {
            name: 'Rumah 2',
            address: 'jalan Mangga 25 blok F nomor 167',
            imgs: [
                {
                    image:
                        'https://res.cloudinary.com/dsvdffre0/image/upload/v1596031267/w3ufcacls7mqfmhbwpe4.png',
                    imgAlt: 'Lakers logo',
                },
                {
                    image:
                        'https://res.cloudinary.com/dsvdffre0/image/upload/v1595015542/sample.jpg',
                    imgAlt: 'Flower',
                },
            ],
            porpertyType: 'apartment',
            saleType: 'rent',
        },
        {
            name: 'Apartment 2',
            address: 'jalan Mangga 25 blok F nomor 167',
            imgs: undefined,
            porpertyType: 'apartment',
            saleType: 'rent',
        },
    ];
    afterEach(cleanup);
    const el: React.ReactElement = <HotList hotProperties={mockProperties} />;

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
