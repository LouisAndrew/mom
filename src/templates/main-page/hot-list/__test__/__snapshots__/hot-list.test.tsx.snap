// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hot List component matches snapshot 1`] = `
<div
  className="sc-fzoXWK TxTJZ"
>
  <div
    className="sc-AxgMl danYDG"
  >
    <h2
      className="sc-AxirZ eWTEun"
      fontFamily="heading"
      fontSize={
        Array [
          4,
          4,
          5,
          5,
        ]
      }
    >
      Lagi hot!! ðŸ”¥ðŸ”¥ðŸ”¥
    </h2>
    <div
      className="sc-fzpmMD jIjIjJ"
      display="flex"
    >
      <div
        className="sc-AxmLO clDedP"
        display="flex"
        overflow="hidden"
        width={
          Array [
            0.9,
            0.45,
            0.4,
          ]
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="sc-fzpans cWuRwg"
            height={
              Array [
                "auto",
              ]
            }
            overflow="hidden"
            width={
              Array [
                1,
              ]
            }
          >
            <img
              alt="Handsome boy"
              src="https://res.cloudinary.com/dsvdffre0/image/upload/v1595340453/ovmhcixqqpfjla4mxovf.jpg"
            />
          </div>
          <div
            className="sc-fzozJi deZYli"
            display="flex"
            width={1}
          >
            <h3
              className="sc-AxiKw dsegVB"
              fontFamily="heading"
              fontSize={
                Array [
                  3,
                  3,
                  4,
                  4,
                ]
              }
            >
              Rumah 1
            </h3>
            <p
              className="sc-AxhUy iPnAqk"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
            >
              jalan Mangga 24 blok G nomor 167
            </p>
            <div
              className="sc-fzoLsD kMKfsv"
              display="flex"
              width={1}
            >
              <span
                className="sc-Axmtr jcmgys"
                color="dark.3"
                fontFamily="body"
                fontSize={
                  Array [
                    1,
                    1,
                    2,
                    2,
                  ]
                }
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={1}
              >
                Jual
              </span>
            </div>
          </div>
        </div>
        <div
          className="sc-fzozJi bngyJH"
          width={1}
        >
          <button
            className="sc-AxheI eICHae"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            onClick={[Function]}
            width={
              Array [
                1,
              ]
            }
          >
            More info
          </button>
        </div>
      </div>
      <div
        className="sc-AxmLO clDedP"
        display="flex"
        overflow="hidden"
        width={
          Array [
            0.9,
            0.45,
            0.4,
          ]
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="sc-fzpans cWuRwg"
            height={
              Array [
                "auto",
              ]
            }
            overflow="hidden"
            width={
              Array [
                1,
              ]
            }
          >
            <img
              alt="Lakers logo"
              src="https://res.cloudinary.com/dsvdffre0/image/upload/v1596031267/w3ufcacls7mqfmhbwpe4.png"
            />
          </div>
          <div
            className="sc-fzozJi deZYli"
            display="flex"
            width={1}
          >
            <h3
              className="sc-AxiKw dsegVB"
              fontFamily="heading"
              fontSize={
                Array [
                  3,
                  3,
                  4,
                  4,
                ]
              }
            >
              Rumah 2
            </h3>
            <p
              className="sc-AxhUy iPnAqk"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
            >
              jalan Mangga 25 blok F nomor 167
            </p>
            <div
              className="sc-fzoLsD kMKfsv"
              display="flex"
              width={1}
            >
              <span
                className="sc-Axmtr hFDSlT"
                color="dark.3"
                fontFamily="body"
                fontSize={
                  Array [
                    1,
                    1,
                    2,
                    2,
                  ]
                }
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={1}
              >
                Sewa
              </span>
            </div>
          </div>
        </div>
        <div
          className="sc-fzozJi bngyJH"
          width={1}
        >
          <button
            className="sc-AxheI eICHae"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            onClick={[Function]}
            width={
              Array [
                1,
              ]
            }
          >
            More info
          </button>
        </div>
      </div>
      <div
        className="sc-AxmLO clDedP"
        display="flex"
        overflow="hidden"
        width={
          Array [
            0.9,
            0.45,
            0.4,
          ]
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="sc-fzpans cWuRwg"
            height={
              Array [
                "auto",
              ]
            }
            overflow="hidden"
            width={
              Array [
                1,
              ]
            }
          >
            <img
              alt="No image provided"
              src="/"
            />
          </div>
          <div
            className="sc-fzozJi deZYli"
            display="flex"
            width={1}
          >
            <h3
              className="sc-AxiKw dsegVB"
              fontFamily="heading"
              fontSize={
                Array [
                  3,
                  3,
                  4,
                  4,
                ]
              }
            >
              Apartment 2
            </h3>
            <p
              className="sc-AxhUy iPnAqk"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
            >
              jalan Mangga 25 blok F nomor 167
            </p>
            <div
              className="sc-fzoLsD kMKfsv"
              display="flex"
              width={1}
            >
              <span
                className="sc-Axmtr hFDSlT"
                color="dark.3"
                fontFamily="body"
                fontSize={
                  Array [
                    1,
                    1,
                    2,
                    2,
                  ]
                }
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={1}
              >
                Sewa
              </span>
            </div>
          </div>
        </div>
        <div
          className="sc-fzozJi bngyJH"
          width={1}
        >
          <button
            className="sc-AxheI eICHae"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            onClick={[Function]}
            width={
              Array [
                1,
              ]
            }
          >
            More info
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
